### PR TITLE
fix(sql): partitioning without designated timestamp should not be possible

### LINKED ISSUE #1726: Create Table should not allow to create partitioned table without designated timestamp
### Describe the bug

_No response_

### To reproduce

```sql
create table pricetable2
(ts timestamp,symbolname SYMBOL CACHE ,price Double,aggregateTradeId Long,firstTradeId Long,lastTradeId Long,buyerMaker Boolean,qty Double) 
partition by day;
```

Executes without error and crates table without designated timestamp partitioned by DAY

### Expected Behavior

Create table query should fail. Partitioning only makes sense on tables with designated timestamps

### Environment

```markdown
- **QuestDB version**:6.1.3
- **OS**:
- **Browser**:
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by puzpuzpuz**
Fixes #1726

A continuation of #1760 (see #1770 to learn why we had to revert this)

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|2|2|100.00%|

**Comment by puzpuzpuz** [comment]
Citation from #1770:
> I will investigate further what exactly caused the crash, but for now the suspicion is that an unordered field was set as designated timestamp, which caused O3 to go wild on memory writes.

I've checked the modified tests and nothing looks suspicious. Test execution times also didn't change. I've run CI four times in a row and didn't notice any CI crashes/failures. So, this one may be ready to go.

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|2|2|100.00%|

File: core/src/test/java/io/questdb/griffin/O3CommitLagTest.java [hunk 1]
```diff
@@ -684,7 +684,7 @@ private void testCommitLagEndingAtPartitionBoundaryPlus10(CairoEngine engine, Sq
         compiler.compile(sql, sqlExecutionContext);
 
         // i=184 is the last entry in date 1970-01-06
-        sql = "create table y as (select * from x where i<=150) partition by DAY";
+        sql = "create table y as (select * from x where i<=150) timestamp(ts) partition by DAY";
```
Comment (bluestreak01) [inline core/src/test/java/io/questdb/griffin/O3CommitLagTest.java:687]: This is very risky. When we move to async filter execution, records might come out of order. `timestamp(ts)` does not enforce order, so there is a risk of creating table with wrong order of records and declared timestamp. O3 will crash.

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

**Review by bluestreak01 — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|2|2|100.00%|

**Comment by bluestreak01** [comment]
@puzpuzpuz this is what we observed before, CI crash

**Comment by puzpuzpuz** [comment]
> @puzpuzpuz this is what we observed before, CI crash

@bluestreak01 
The recent Windows CI crash seems to be related with `LineTcpReceiverFuzzTest#testLoad` and system-wise OOM:
```
# There is insufficient memory for the Java Runtime Environment to continue.
# Native memory allocation (malloc) failed to allocate 941856 bytes for Chunk::new
# An error report file with more information is saved as:
# D:\a\1\s\core\hs_err_pid5668.log
```

Was it the same when these changes weren't reverted yet?

**Comment by bluestreak01** [comment]
I don't remember in fact :( Regardless, I think it is risky to merge considering this is a first instance of a crash here since ILP fixes

**Review by bluestreak01 — CHANGES_REQUESTED** [review]
We will have to understand the crash before merging

**Comment by puzpuzpuz** [comment]
> I don't remember in fact :( Regardless, I think it is risky to merge considering this is a first instance of a crash here since ILP fixes

OK, let's see if the same crash shows up on other branches. It may be unrelated to the changes after all.

**Comment by bluestreak01** [comment]
> OK, let's see if the same crash shows up on other branches. It may be unrelated to the changes after all.

Cool, lets trigger other branches manually if need be.

**Comment by ideoma** [comment]
I saw `Native memory allocation (malloc) failed to allocate` on other builds without this change on Windows. @glasstiger and I came to conclusion that it was genuine OS out of memory.

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlParser.java|2|2|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Comment by puzpuzpuz** [comment]
@kocko many thanks for the contribution! Sorry for the back-and-forth with merging your patch.

**Pull request merged**